examples/sat/hillclimber.ipynb

```diff
@@ -0,0 +1,757 @@
+{
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import numpy as np\n",
+    "import pandas as pd\n",
+    "from IPython.display import display, Math\n",
+    "import logging\n",
+    "\n",
+    "for h in logging.getLogger().handlers:\n",
+    "    h.setFormatter(logging.Formatter(\"[%(levelname)s] %(message)s\"))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Benchmarking an algorithm used to solve a MAX-k-SAT problem\n",
+    "\n",
+    "In this notebook we showcase functionality of qubrabench by implementing and benchmarking a hillclimbing algorithm to solve MAX-k-SAT, as described in https://arxiv.org/abs/2203.04975.\n",
+    "<!-- The paper describes two distinct variants of implementing the hillclimber: a simple hillclimber - which uses (quantum) search, and a steep one - which uses (quantum) max finding. -->\n",
+    "\n",
+    "## Problem: MAX-k-SAT\n",
+    "\n",
+    "Max-k-SAT is a combinatorial optimization problem that given a list of clauses $(C_{i})^{p}_{i=1}$, each a disjunction of at most $k$ literals, and a set of weights $(w_{i})^{p}_{i=1}$, asks us to maximize the weight of the satisfied clauses,\n",
+    "$$\\varphi(x) := \\sum ^{p} _{i=1} w_{i}C_{i}(x),$$\n",
+    "over all assignments $x \\in \\{0, 1\\}^{q}$ of the variables. This problem is known to be NP-hard for $k ≥ 2$."
```

Comment:
I believe this is a lingering typo and it should be NP-hard for $k ≥ 3$

Reply:
No, MAX-k-SAT is hard for k >= 2.